app-android/src/main/java/com/ustadmobile/port/android/view/ContentEntryListRecyclerAdapter.kt

```diff
@@ -79,8 +89,12 @@ class ContentEntryListRecyclerAdapter(var itemListener: ContentEntryListItemList
         holder.itemView.setSelectedIfInList(item, selectedItems, ContentEntryList2Fragment.DIFF_CALLBACK)
         if(item != null) {
             GlobalScope.launch(Dispatchers.Main.immediate) {
-                holder.downloadJobItemLiveData = containerDownloadManager
-                        .getDownloadJobItemByContentEntryUid(item.contentEntryUid )
+                holder.downloadJobItemLiveData = RateLimitedLiveData(appDatabase, listOf("ContentJobItem"), 1000) {
```

Comment:
There is no need to use GlobalScope.launch here.

Reply:
This needs to handle when item is null, eg if the item is null, then holder.downloadJobItemLiveData should be set to null.